### PR TITLE
Rework expression processor

### PR DESCRIPTION
**Pull request by juliendbg**
- improve expression language documentation
- add forced field name

### CODE REVIEW

File: source/publishing_data/05_processing_data/processors/expression.rst [hunk 1]
```diff
@@ -1,273 +1,347 @@
...
+     * * ``2 + 3`` returns ``5``
+       * ``5 - 8`` returns ``-3``
+       * ``9 * 3`` returns ``27``
+       * ``15 / 6`` returns ``2.5``
+       * ``15 % 6`` returns ``3``
+       * ``15 // 6`` returns ``2``
+       * ``2 ^ 3`` returns ``8``
+       * ``column_1 + column2``
+   * * ``and``, ``&&``, ``or``, ``||`` (return a boolean)
+     * Boolean operators: and, and, or, or
```
Comment (clement-tourriere) [inline source/publishing_data/05_processing_data/processors/expression.rst]: A little strange to duplicate this. Don't we want a list instead ? Is it necessary to speak about binary operator priority ? e.g.: if you have a || and a && in the same expression, what is the operator priority.

Comment (juliendbg) [reply source/publishing_data/05_processing_data/processors/expression.rst]: Indeed, I may have talked about operator precedence. Let's add it in another change so this one can get translated and published.

Comment (juliendbg) [reply source/publishing_data/05_processing_data/processors/expression.rst]: Done

File: source/publishing_data/05_processing_data/processors/expression.rst [hunk 2]
```diff
@@ -1,273 +1,352 @@
...
-     * center, datediff, ljust, replace, rjust, substring
+   * * Operator
+     * Description
+     * Example
+   * * ``-``
+     * Prefix that negates the following value
+     * ``- 4``, ``- [expression]``
+   * * ``not``, ``!`` (not)
+     * Boolean operator that inverts the following condition
+     * * ``not true`` or  ``!true`` returns ``false``
```
Comment (vbarbaresi) [inline source/publishing_data/05_processing_data/processors/expression.rst]: `not` doesn't work like an operator but as a method: `not('x')`

same usage as `!()` that is correct 2 lines below

File: source/publishing_data/05_processing_data/processors/expression.rst [hunk 3]
```diff
@@ -1,273 +1,352 @@
...
 
 .. admonition:: Note
    :class: note
 
-   Keep in mind that for each expression, the examples are presented with actual numbers, but they can be replaced with the processed dataset's fields technical identifiers.
+   A ternary expression can be read as "if [condition] then [result if true] else [result if false]".
+
+Examples:
+
+- ``true ? 'hello' : 'goodbye'`` returns ``'hello'``
```
Comment (vbarbaresi) [inline source/publishing_data/05_processing_data/processors/expression.rst]: the examples with ` ``true`` ` need quotes around true/false to work; ` ``'true'`` `

File: source/publishing_data/05_processing_data/processors/expression.rst (hunk 2 above)
Comment (juliendbg) [reply source/publishing_data/05_processing_data/processors/expression.rst]: Good catch 👌

File: source/publishing_data/05_processing_data/processors/expression.rst (hunk 3 above)
Comment (juliendbg) [reply source/publishing_data/05_processing_data/processors/expression.rst]: 👍

**Pull request merged**